systemtests/src/main/java/io/enmasse/systemtest/TestUtils.java

```diff
@@ -391,15 +391,15 @@ public static String getAddressSpaceType(JsonObject addressSpace) {
      * @param addressSpace name of addressSpace
      * @throws Exception IllegalStateException if address space is not ready within timeout
      */
-    public static AddressSpace waitForAddressSpaceReady(AddressApiClient apiClient, String addressSpace) throws Exception {
+    public static AddressSpace waitForAddressSpaceReady(AddressApiClient apiClient, AddressSpace addressSpace) throws Exception {
         JsonObject addressSpaceObject = null;
         TimeoutBudget budget = null;
 
         boolean isReady = false;
         budget = new TimeoutBudget(5, TimeUnit.MINUTES);
         while (budget.timeLeft() >= 0 && !isReady) {
-            addressSpaceObject = apiClient.getAddressSpace(addressSpace);
-            isReady = isAddressSpaceReady(addressSpaceObject);
+            addressSpaceObject = apiClient.getAddressSpace(addressSpace.getName());
+            isReady = isAddressSpaceReady(addressSpaceObject) && matchAddressSpacePlan(addressSpaceObject, addressSpace);
```

Comment:
For tests that want to test that a plan cannot be applied (due to too little quota), I think it might make sense to split this into a waitForAddressSpaceReady and waitForAddressSpacePlanApplied or something like that, and invoke them separately like https://github.com/EnMasseProject/enmasse/pull/2098/files#diff-3d1b3eed8fb655a202d75bab060610d0R304

If you write a test that wants to verify that plan is not applied, splitting them up would make it possible to provide more context about why the test failed.